internal/interface/httpapi/api.go

```diff
@@ -73,36 +72,176 @@ type getMusicalEntityResponseModel struct {
 }
 
 func (a *Api) getMusicalEntity(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	eid, ok := vars[entityIdUrlPathKey]
+	if !ok {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	e, err := a.MusicalEntityUseCases.GetMusicalEntityById(eid)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
 	m := getMusicalEntityResponseModel{
-		Artist: "Radiohead", Album: "Hail To the Thief", Track: "2 + 2 = 5",
-		Links: []link{link{ServiceName: "Yandex Music", Url: "https://music.yandex.ru/track/333416"}},
+		Artist: e.Artist,
+		Album: e.Album,
+		Track: e.Track,
+		Links: make([]link, 0, len(e.Links)),
+	}
+	for _, l := range e.Links {
+		m.Links = append(m.Links, link{
+			ServiceName: l.ServiceName,
+			Url: l.Url,
+		})
 	}
 	if err := json.NewEncoder(w).Encode(m); err != nil {
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
 }
 
-func (a *Api) putMusicalEntity(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusNotImplemented)
-}
-
 type getPlaylistResponseModel struct {
-	Content  []getMusicalEntityResponseModel `json:"content"`
+	Name    string                          `json:"name"`
+	Content []getMusicalEntityResponseModel `json:"content"`
 }
 
 func (a *Api) getPlaylist(w http.ResponseWriter, r *http.Request) {
-	w.WriteHeader(http.StatusNotImplemented)
+	vars := mux.Vars(r)
+	pid, ok := vars[playlistIdUrlPathKey]
+	if !ok {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	p, err := a.PlaylistUseCases.GetPlaylistById(pid)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	m := getPlaylistResponseModel{
+		Name: p.Name,
+		Content: make([]getMusicalEntityResponseModel, 0, len(p.Content)),
+	}
+	for _, c := range p.Content {
```

Comment:
It's time to introduce a function here: too many nested loops are always dangerous.